exterminate.lua

```diff
@@ -0,0 +1,206 @@
+    {'o', 'only-visible', handler = function() options.only_visible = true end},
+    {'e', 'only-hostile', handler = function() options.only_hostile = true end},
+})
+
+if not dfhack.isMapLoaded() then
+    qerror('This script requires a fortress map to be loaded')
+end
+
+if positionals[1] == "help" or options.help then
+    print(dfhack.script_help())
+end
+
+if positionals[1] == "this" then
+    local selected_unit = dfhack.gui.getSelectedUnit()
+
+    if not selected_unit then
+        qerror("Select a unit and run the script again.")
+    end
+
+    killUnit(selected_unit)
+end
+
+if positionals[1] == nil then
+    local map_races = getMapRaces(options.only_visible)
+
+    local sorted_races = {}
+    for race, value in pairs(map_races) do
+        table.insert(sorted_races, { name = race, count = value.count })
+    end
+
+    table.sort(sorted_races, function(a, b)
+        return a.count > b.count
+    end)
+
+    for _, race in pairs(sorted_races) do
+        print(([[%4s %s]]):format(race.count, race.name))
+    end
+
+    return
+end
+
+local map_races = getMapRaces(options.only_visible)
+
+if string.find(positionals[1], "UNDEAD") then
+    if map_races.UNDEAD then
+        for _, unit in pairs(df.global.world.units.active) do
+            if dfhack.units.isUndead(unit) and checkUnit(unit) then
+                killUnit(unit)
+            end
+        end
+    else
+        qerror("No undead found on the map.")
+    end
+else
+    local selected_race, selected_caste = positionals[1], nil
+
+    if string.find(selected_race, ':') then
+        local tokens = positionals[1]:split(':')
+        selected_race, selected_caste = tokens[1], tokens[2]
+    end
+
+    if not map_races[selected_race] then
+        qerror("No creatures of this race on the map.")
+    end
+
+    local race_castes = getRaceCastes(map_races[selected_race].id)
+
+    if selected_caste and not race_castes[selected_caste] then
+        qerror("Invalid caste.")
+    end
+
+    local count = 0
+    for _, unit in pairs(df.global.world.units.active) do
+        if not dfhack.units.isActive(unit) or not checkUnit(unit) then
+            goto skipunit
+        end
+
+        if selected_caste and selected_caste ~= df.creature_raw.find(unit.race).caste[unit.caste].caste_id then
+            goto skipunit
+        end
+
+        if selected_race == df.creature_raw.find(unit.race).creature_id then
+            if options.method == killMethod.BUTCHER then
+                butcherUnit(unit)
+            elseif options.method == killMethod.MAGMA then
+                drownUnit(unit, df.tile_liquid.Magma)
+            elseif options.method == killMethod.DROWN then
+                drownUnit(unit, df.tile_liquid.Water)
+            else
+                killUnit(unit)
+            end
+
+            count = count + 1
+        end
+
+        :: skipunit ::
+    end
+
+    print(([[Exterminated %s creatures.]]):format(count))
```

Comment:
not that there's anything specifically wrong with it, but why did you choose the `[[` `]]` format?

Reply:
More or less habit, do you find it not quite as readable? When using :format I tend to gravitate over to it.

Reply:
no, it's totally fine. I was just curious.